### PR TITLE
fix(curriculum): add example for arrow functions in Platformer Game

### LINKED ISSUE #54956: Add example for Destructuring assignment within arrow functions in platformer project
### Describe the Issue

Destructuring is explained in earlier challenges but primarily when 'automatically' or bulk copying object properties , to a list of variables in curly brackets. This question although similar has this being done in an arrow functions input. I spent around 2 hours trying to get to the bottom of this question, when in reality the answer was fairly straight forward once principles were understood in relation to this different context (arrow functions parameters). 

Maybe some tips or an example would of helped, I realise that you are trying to enable people to do these operations without prompting, but usually a few examples are given when a topic is moderately new or complex. I felt there was strong indication in question to construct an answer that conformed to the simpler type of destructuring, without too much indication, that this was wrong, only very careful reading of questions grammar might lead you to understand that the answer lay in the functions parameter code, not the body.

A brief example or few sentences would of made alot of difference to my experience, which is normally great.

I have been asked to construct this issue in response to help received on the forum. I have put the code that I thought erroneously was how to do this , along with correct method

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-intermediate-oop-by-building-a-platformer-game/step-59

### Your code

Wrong way....
`window.addEventListener("keydown", (event) => {  {key}}= event});`

Where this is correct answer ....

`window.addEventListener("keydown", ({ key }) => { `
});

### Expected behavior

Bit more help , thats all, there is no coding error

### Screenshots

_No response_

### System

- Device:  Laptop
- OS: Windows 10
- Browser: Chrome
- Version: 124

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aliarnoos**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54956

Summary: 

added an example function in how to use an arrow function with destruction of its parameters for **step 59** of the course **JavaScript Algorithms and Data Structures (Beta)**

### CODE REVIEW

**Comment by aliarnoos** [comment]
This is my first contribution, so I would appreciate any feedback if I mess up any step of the process, or any feedback in general. Thank you :)

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c72e52133d687e8e6a60f6.md [hunk 1]
```diff
@@ -13,6 +13,12 @@ Start by adding an `addEventListener` to the global `window` object.
 
 For the arguments, pass in the `keydown` event and an arrow function that uses the destructuring assignment to get the `key` property from the `event` object in the event listener parameter. 
 
+```js
+btn.addEventListener('click', ({example}) => {
+  console.log(example);
+});
+```
+
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c72e52133d687e8e6a60f6.md:21]:
````suggestion
Here is the syntax for using the destructuring assignment in the parameter list of the arrow function:

```js
btn.addEventListener('click', ({example}) => {
  console.log(example);
});
```

````

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Thanks for your interest in contributing! We'll need to add the text explaining that the code is an example.

**Comment by naomi-lgbt** [comment]
As an aside, you should always work off of a new branch and not your `main` branch.

**Comment by aliarnoos** [comment]
Noted!, thank you so much

**Review by naomi-lgbt — APPROVED** [review]

**Comment by lasjorg** [comment]
I'm not crazy about the example code. Considering it is the event object, and it won't have an `example` property on that object. So if they test it and run the code, it will just log `undefined`.

I would suggest using a more "real" example. Or at least some that works/does something.
```
const user = { id: 42, name: "arthur" };

function getUserId({ id }) {
  console.log(id); // 42
}

getUserId(user);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c72e52133d687e8e6a60f6.md [hunk 2]
```diff
@@ -13,6 +13,14 @@ Start by adding an `addEventListener` to the global `window` object.
 
 For the arguments, pass in the `keydown` event and an arrow function that uses the destructuring assignment to get the `key` property from the `event` object in the event listener parameter. 
 
+Here is the syntax for using the destructuring assignment in the parameter list of the arrow function:
+
+```js
+btn.addEventListener('click', ({example}) => {
+  console.log(example);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c72e52133d687e8e6a60f6.md:20]:
```suggestion
btn.addEventListener('click', ({ target }) => {
  console.log(target);
```

Comment (lasjorg) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c72e52133d687e8e6a60f6.md:20]: @aliarnoos please make the suggested change.

**Review by lasjorg — APPROVED** [review]
Looks good to me.

**Pull request merged**